tests/test_layers.py

```diff
@@ -223,6 +223,25 @@ def model_func(wrapped_import, inputs: torch.Tensor):
   verify_torch_and_convert_to_returnn(model_func, inputs=x)
 
 
+def test_reshape():
+  n_batch, n_feature, n_time = 3, 8, 5
+
+  def model_func_1(wrapped_import, inputs: torch.Tensor):
+    # test case (..., a*b, c,...) -> (..., a, b*c,...)
+    return inputs.view(n_batch, n_feature // 2, n_time * 2)
```

Comment:
This is not exactly correct for the time dim. On RETURNN side, `inputs` has shape [B,D,T] where T is dynamic. Then you make it static here. This might break at some point. I actually wonder that it works without error now.

I would introduce another second static axis for `inputs`, and do this reshape test on two static axes (not involving time or batch).

When you want to test specifically for correct handling of time or batch, make a separate test for it (but maybe also separate PR, so we can already merge this, and then do further work on it).